src/libs/API.js

```diff
@@ -159,15 +156,12 @@ Network.registerResponseHandler((queuedRequest, response) => {
         // There are some API requests that should not be retried when there is an auth failure like
         // creating and deleting logins. In those cases, they should handle the original response instead
         // of the new response created by handleExpiredAuthToken.
-        const shouldRetry = lodashGet(queuedRequest, 'data.shouldRetry');
-        if (!shouldRetry || unableToReauthenticate) {
+        if (!shouldRetry(queuedRequest) || unableToReauthenticate) {
             queuedRequest.resolve(response);
             return;
         }
 
-        handleExpiredAuthToken(queuedRequest.command, queuedRequest.data, queuedRequest.type)
-            .then(queuedRequest.resolve)
-            .catch(queuedRequest.reject);
+        handleExpiredAuthToken(queuedRequest);
```

Comment:
Should we pause the request queue right here rather than in `Authenticate`? Because we know on line 55 if we're going to try to reauthenticate, so maybe we should immediate pause the request queue so that another request doesn't begin processing before we attempt to reauthenticate?